src/Microsoft.ML.TimeSeries/Forecast.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.ML.Data;
+using static Microsoft.ML.Transforms.TimeSeries.AdaptiveSingularSpectrumSequenceModeler;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Interface for forecasting models.
+    /// </summary>
+    /// <typeparam name="T">The type of values that are forecasted.</typeparam>
+    public interface ICanForecast<T> : ICanSaveModel
+    {
+        /// <summary>
+        /// Train a forecasting model from an <see cref="IDataView"/>.
+        /// </summary>
+        /// <param name="dataView">Reference to the <see cref="IDataView"/></param>
+        void Train(IDataView dataView);
+
+        /// <summary>
+        /// Update a forecasting model with the new observations in the form of an <see cref="IDataView"/>.
+        /// </summary>
+        /// <param name="dataView">Reference to the observations as an <see cref="IDataView"/></param>
+        /// <param name="inputColumnName">Name of the input column to update from.</param>
+        void Update(IDataView dataView, string inputColumnName = null);
+
+        /// <summary>
+        /// Perform forecasting until a particular <paramref name="horizon"/>.
+        /// </summary>
+        /// <param name="horizon">Number of values to forecast.</param>
+        /// <returns>Forecasted values.</returns>
+        T[] Forecast(int horizon);
+
+        /// <summary>
+        /// Perform forecasting until a particular <paramref name="horizon"/> and also computes confidence intervals.
+        /// </summary>
+        /// <param name="horizon">Number of values to forecast.</param>
+        /// <param name="forecast">Forecasted values</param>
+        /// <param name="confidenceIntervalLowerBounds">Lower bound confidence intervals of forecasted values.</param>
+        /// <param name="confidenceIntervalUpperBounds">Upper bound confidence intervals of forecasted values.</param>
+        /// <param name="confidenceLevel">Confidence level.</param>
```

Comment:
>Confidence level. [](start = 42, length = 17)

Confidence of what? This doc string doesn't say more than the associated variable name.